core/src/apps/bitcoin/scripts.py

```diff
@@ -573,9 +628,14 @@ def read_bip322_signature_proof(r: utils.BufferReader) -> tuple[memoryview, memo
 
 
 def write_signature_prefixed(w: Writer, signature: bytes, hash_type: int) -> None:
-    write_bitcoin_varint(w, len(signature) + 1)
+    length = len(signature)
+    if hash_type != common.SIGHASH_ALL_TAPROOT:
```

Comment:
How am I supposed to cast `int` to `SigHashType`? E.g. if I do `sighash_type = SigHashType(r.get())`, then mypy is happy but the code is broken.

Reply:
where is the sighash_type coming in as int?

Reply:
is it some reader? in full Python there's `typing.cast(SigHashType, r.get())`
for us, i think this would work:
```
sighash_type: SigHashType
sighash_type = r.get()  # type: ignore
```

Reply:
...but given that this usage exists, i.e., that sighash type can come as an int from outside, maybe it's not such a good idea to enforce-but-not-really the type

Reply:
https://github.com/trezor/trezor-firmware/pull/1891/commits/8ac8217c1637b8651043e45a503da95cc8aca670

Reply:
very nice!